### PR TITLE
executorch export docs: input contiguity

### LINKED ISSUE #1233: ExecuTorch XNNPACK `.pte` returns 0 detections on real images
**ExecuTorch XNNPACK `.pte` returns 0 detections on real images (backbone diverges from eager)**

## Summary

`RFDETRSmall().export(format="executorch", backend="xnnpack")` produces a `.pte` whose inference **diverges massively from eager PyTorch on real images**, collapsing every detection (all confidences < 0.25, all one class) → **0 detections**. It looks correct only because the existing parity test feeds **random** input, where the divergence does not appear.

- `predict()` on `dog.jpeg`: 4 detections `[0.95, 0.66, 0.58, 0.53]`
- eager **export-mode** forward, same image: same 4 detections `[0.96, 0.63, 0.58, 0.52]`
- XNNPACK `.pte`, same image: `[0.247, 0.241, 0.222, 0.208, 0.193]`, labels `[1,1,1,1,1]` → **0 above 0.5**

## Where the bug is (bisected)

`torch.export`'s ExportedProgram is **bit-exact** with eager; only the executorch-lowered `.pte` is wrong:

| Comparison | max abs delta logits |
| --- | --- |
| eager vs `torch.export` program (pre-lowering) | **0.0** |
| eager vs XNNPACK `.pte` | **10.47** |

Stage-by-stage inside the `.pte` (real image), divergence starts at the **DINOv2 windowed-attention backbone** and cascades:

| stage | shape | max abs delta (eager vs pte) |
| --- | --- | --- |
| backbone output | 1x256x32x32 | **3.09** (origin) |
| two-stage memory | 1x300x256 | 3.49 |
| decoder | 1x300x256 | 5.40 |
| pred_logits | 1x300x91 | 10.47 |

Ruled out:
- **`topk`/`max.dim`**: standalone through XNNPACK on the real scores gives 300/300 exact match (so the QNN-style index-divergence fix does not apply to XNNPACK).
- **`torch.export`**: perfect (0.0).
- **Regression**: reproduced at the #1142 merge commit too — present since the feature landed.

Input-dependence (the reason the current test misses it):

| input | max abs delta logits |
| --- | --- |
| `torch.randn` | 3e-4 |
| `torch.ones*0.5` | 1e-4 |
| real image (`dog.jpeg`, ImageNet-normalized) | **10.47** |

Portable executorch (no XNNPACK delegation) is **not** a workaround — it crashes on the patch-embed conv: `op_convolution.cpp: check_convolution_args failed` (kernel 16x16, stride 16, 3->384).

## Environment

- clean venv, macOS arm64, Python 3.12
- `torch==2.12.1`, `executorch==1.3.1` (the pair rfdetr validates), `flatc` 25.x
- also reproduced on Colab (Linux, `torch==2.12.1+cu130`)

## Repro

```python
import torch
from PIL import Image
from rfdetr import RFDETRSmall
from rfdetr.export.benchmark import infer_transforms, post_process

image = Image.open("dog.jpeg").convert("RGB")  # any real photo with COCO objects
model = RFDETRSmall(); res = model.model_config.resolution

pte = model.export(format="executorch", backend="xnnpack", output_dir="out")

core = model.model.model.eval().cpu(); core.export()          # export-mode reference
t, _ = infer_transforms((res, res))(image, None); x = t[None].float()
with torch.no_grad(): eager = core(x.clone())

from executorch.runtime import Runtime
pb, pl = Runtime.get().load_program(str(pte)).load_method("forward").execute([x.clone()])

print("max|eager-pte| logits:", (eager[1] - pl).abs().max().item())   # ~10.5
ts = torch.tensor([[image.height, image.width]])
print("eager >0.5:", int((post_process({'dets':eager[0],'labels':eager[1]}, ts)[0]['scores'] > 0.5).sum()))  # 4
print("pte   >0.5:", int((post_process({'dets':pb,'labels':pl}, ts)[0]['scores'] > 0.5).sum()))              # 0
```

(`.pte` export needs the `flatc` binary on PATH — `brew install flatbuffers` / `apt install flatbuffers-compiler`.)

## Impact

- XNNPACK detection export is unusable on real images (the documented, pip-installable default backend).
- The `.pte` **export** succeeds and is numerically faithful for random input, so nothing errors — it silently returns wrong detections.

## Suggested actions

1. **Change the parity test to use a real (or structured) image, not `torch.randn`** — random input does not exercise the failing path. This is the coverage gap that let it ship.
2. Reproduce against a newer/nightly `executorch` to see if the XNNPACK delegate bug is fixed upstream; bump the pin if so.
3. If worked around in-repo: identify the XNNPACK-delegated backbone op that mis-computes and exclude it from `XnnpackPartitioner` (portable fallback currently blocked by the conv arg-check failure above).
4. Until fixed, document XNNPACK detection inference as known-broken; steer users to CoreML (Apple) for on-device detection.

### PR DESCRIPTION
**Pull request by anatoly-ryabchenko**
## What does this PR do?

**Related Issue(s):** closes #1233 

## Type of Change

- Documentation update
+ new tests

## Testing

- [x] I have tested this change locally
- [x] I have added/updated tests for this change

**Test details:**
Added 2 tests: 
- parity with a real image (postprocessed detections instead of raw outputs, since they do diverge slightly, but that is expected) and 
- infer_transform input is contiguous

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have updated the documentation accordingly (if applicable)

### CODE REVIEW

**Comment by Borda** [comment]
@anatoly-ryabchenko, so can we not add in the cookbook also inference on the CPU machine to show output detection?

**Comment by anatoly-ryabchenko** [comment]
> @anatoly-ryabchenko, so can we not add in the cookbook also inference on the CPU machine to show output detection?

Done

**Review by Borda — DISMISSED** [review]
